test/cases/ostree-vsphere.sh

```diff
@@ -0,0 +1,884 @@
+
+# Prepare the blueprint for the compose.
+greenprint "📋 Preparing installer blueprint"
+sudo composer-cli blueprints push "$BLUEPRINT_FILE"
+sudo composer-cli blueprints depsolve vmdk
+
+# Build simplified installer iso image.
+build_image vmdk "${INSTALLER_TYPE}" "${PROD_REPO_URL}/"
+
+# Download the image
+greenprint "📥 Downloading the vmdk image"
+sudo composer-cli compose image "${COMPOSE_ID}" > /dev/null
+ISO_FILENAME="${COMPOSE_ID}-${INSTALLER_FILENAME}"
+sudo mv "${ISO_FILENAME}" edge-vsphere.vmdk
+
+# Clean compose and blueprints.
+greenprint "🧹 Clean up vmdk blueprint and compose"
+sudo composer-cli compose delete "${COMPOSE_ID}" > /dev/null
+sudo composer-cli blueprints delete vmdk > /dev/null
+
+##################################################################
+##
+## Create vm on datacenter7.0-amd and test it
+##
+##################################################################
+# Datacenter setup
+VSPHERE_DC70="Datacenter7.0-AMD"
+VSPHERE_DC70_HOST="10.73.196.21"
+VSPHERE_DC70_DATASTORE="datastore-21"
+DC70_VSPHERE_VM_NAME="${COMPOSE_ID}-70"
+
+# Remove all existing disk files
+greenprint "🧹 Remove existing vmdk disk files in vsphere datacenter"
+sudo sshpass -p "${DATACENTER_PASSWORD}" ssh "${SSH_OPTIONS[@]}" "${DATACENTER_USERNAME}@${VSPHERE_DC70_HOST}" \
+    rm -fr /vmfs/volumes/${VSPHERE_DC70_DATASTORE}/edge-vsphere
+
+# Upload image to vsphere and create VM
+export GOVC_URL="https://bootp-73-75-144.lab.eng.pek2.redhat.com/sdk"
+export GOVC_USERNAME="${VSPHERE_USERNAME}"
+export GOVC_PASSWORD="${VSPHERE_PASSWORD}"
+export GOVC_INSECURE="true"
+
+greenprint "📋 Uploading vmdk image to vsphere datacenter 7.0"
+export GOVC_DATACENTER="Datacenter7.0-AMD"
+govc import.vmdk -ds="datastore-21" -pool="/Datacenter7.0-AMD/host/Cluster7.0-AMD/Resources" ./edge-vsphere.vmdk
+
+# Create vm with vmdk image
+greenprint "📋 Create vm in vsphere datacenter"
+sudo ansible-playbook -v -i "${TEMPDIR}"/inventory \
+    -e vsphere_host="${VSPHERE_HOST}" \
+    -e vsphere_username="${VSPHERE_USERNAME}" \
+    -e vsphere_password="${VSPHERE_PASSWORD}" \
+    -e vsphere_datacenter="${VSPHERE_DC70}" \
+    -e vsphere_esxi_host="${VSPHERE_DC70_HOST}" \
+    -e vsphere_datastore="${VSPHERE_DC70_DATASTORE}" \
+    -e vsphere_vm_name="${DC70_VSPHERE_VM_NAME}" \
+    /usr/share/tests/osbuild-composer/ansible/vmdk-create-vm.yaml || RESULTS=0
+
+DC70_GUEST_ADDRESS=$(cat /usr/share/tests/osbuild-composer/ansible/vmdk_ip.txt)
+sudo rm -fr /usr/share/tests/osbuild-composer/ansible/vmdk_ip.txt
+greenprint "🛃 Edge VM IP address is: ${DC70_GUEST_ADDRESS}"
+
+# Run ansible check on edge vm
+# Check for ssh ready to go.
+greenprint "🛃 Checking for SSH is ready to go"
+for _ in $(seq 0 30); do
+    RESULTS="$(wait_for_ssh_up "${DC70_GUEST_ADDRESS}")"
+    if [[ $RESULTS == 1 ]]; then
+        echo "SSH is ready now! 🥳"
+        break
+    fi
+    sleep 10
+done
+
+# With new ostree-libs-2022.6-3, edge vm needs to reboot twice to make the /sysroot readonly
```

Comment:
Only edge-commit and edge-container needs this kind of reboot.

Reply:
Removed reboot step.